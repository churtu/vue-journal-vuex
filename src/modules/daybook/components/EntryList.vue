<template>
  <div class="entry-list-container">
      <div class="px-2 pt-2">
          <input v-model="term" type="text" class="form-control" placeholder="Buscar entrada">
      </div>
      <div class="mt-2 d-flex flex-column">
          <button @click="$router.push({name: 'entry', params: {id: 'new'}})" class="btn btn-primary mx-3">Nueva entrada <i class="fa fa-plus-circle"></i></button>
      </div>
      <div class="entry-scrollarea">
          <entry
          v-for="entry in entriesbyTerm" :key="entry.id"
          :entry="entry"
          />

      </div>
  </div>
</template>

<script>
import { defineAsyncComponent } from 'vue'
import { mapGetters } from 'vuex'

export default {
    components:{
        Entry: defineAsyncComponent(() => import('../components/Entry'))
    },
    computed: {
        ...mapGetters('journal', ['getEntriesByTerm']),
        entriesbyTerm(){
            return this.getEntriesByTerm( this.term )
        }
    },
    data(){
        return{
            term: ''
        }
    }
}
</script>

<style lang="scss" scoped>
    .entry-list-container{
        border-right: 1px solid #2c3e50;
        height: calc(100vh - 56px);
    }

    .entry-scrollarea{
        height: calc(100vh - 110px);
        overflow: auto;
    }
</style>