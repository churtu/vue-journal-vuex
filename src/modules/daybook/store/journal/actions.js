import journalAPI from '@/api/journalAPI'


export const loadEntries = async ( { commit }) => {
    const { data } = await journalAPI.get('/entries.json')
    if(!data){
        commit('setEntries', [])
        return
    }
    const entries = []
     
    for( let id of Object.keys( data )){
        entries.push({
            id,
            ...data[id]
        })
    }
    commit('setEntries', entries)
}

export const updateEntry = async ( { commit }, entry) =>{
    const { picture, date, text, id } = entry 
    const dataToSave = {date, text, picture}
    await journalAPI.put(`/entries/${id}.json`, dataToSave)
    dataToSave.id = id
    commit('updateEntry', { ...dataToSave })
    
}

export const createEntry = async ( { commit }, entry ) => {
    const { text, date, picture } = entry
    const dataToSave = {date, text, picture}
    const { data } = await journalAPI.post('/entries.json', dataToSave)
    commit('addEntry', {text, date, id: data.name})
    return data.name // id en firebase
}

export const deleteEntry = async ( { commit }, id ) => {
    await journalAPI.delete(`/entries/${id}.json`)
    commit('deleteEntry', id)
}