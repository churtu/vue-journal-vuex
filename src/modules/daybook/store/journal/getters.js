
// export const myGetter = ( /*state*/ ) => {

// }

export const getEntiesByTerm = ( /*state*/ ) => {

}

export const getEntryById = ( /*state*/ ) => {

}