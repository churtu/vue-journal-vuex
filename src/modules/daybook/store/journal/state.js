

export default () => ({
    isLoading:true,
    entries: [
        {
            id: new Date().getTime(),
            date: new Date().toDateString(),
            text: 'Voluptate consectetur minim quis esse duis laborum dolore Lorem in cupidatat aliquip consequat nisi amet. Adipisicing nisi consequat consequat ad amet commodo. Eiusmod id laboris consequat aliqua veniam consectetur tempor. Officia aliquip veniam pariatur nostrud aliquip labore ad Lorem sint ea officia. Sint velit quis tempor excepteur occaecat est pariatur.',
            picture: null
        },
        {
            id: new Date().getTime()+1000,
            date: new Date().toDateString(),
            text: 'Voluptate consectetur minim quis esse duis laborum dolore Lorem in cupidatat aliquip consequat nisi amet. Adipisicing nisi consequat consequat ad amet commodo. Eiusmod id laboris consequat aliqua veniam consectetur tempor. Officia aliquip veniam pariatur nostrud aliquip labore ad Lorem sint ea officia. Sint velit quis tempor excepteur occaecat est pariatur.',
            picture: null
        },
        {
            id: new Date().getTime()+2000,
            date: new Date().toDateString(),
            text: 'Voluptate consectetur minim quis esse duis laborum dolore Lorem in cupidatat aliquip consequat nisi amet. Adipisicing nisi consequat consequat ad amet commodo. Eiusmod id laboris consequat aliqua veniam consectetur tempor. Officia aliquip veniam pariatur nostrud aliquip labore ad Lorem sint ea officia. Sint velit quis tempor excepteur occaecat est pariatur.',
            picture: null
        },
    ]
})