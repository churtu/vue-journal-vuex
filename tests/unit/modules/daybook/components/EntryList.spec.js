import { shallowMount } from '@vue/test-utils'
import { createStore } from 'vuex'
import { journalState } from '../../../mock-data/test-journal-state'
// import { getEntriesByTerm } from '@/modules/daybook/store/journal/getters'
import EntryList from '@/modules/daybook/components/EntryList'

import journal from '@/modules/daybook/store/journal'

const createVuexStore = ( initialState ) => createStore({
    modules: {
        journal: {
            ...journal,
            state: initialState
        }
    }
})

describe('Pruebas en el EntryList', () => {

    // const journalMockModule = {
    //     namespaced: true,
    //     getters:{
    //         getEntriesByTerm
    //     },
    //     state: () => ({
    //         isLoading: false,
    //         entries: journalState.entries
    //     })
    // }
    // const store = createStore({
    //     modules: {
    //         journal: { ...journalMockModule }
    //     }
    // })

    // const wrapper = shallowMount(EntryList, {
    //     global: {
    //         mocks: {
    //             $router: routerMock
    //         },
    //         plugins: [ store ]
    //     },
    // })

    const routerMock = {
        push: jest.fn()
    }

    const store = createVuexStore( journalState )

    let wrapper 

    beforeEach(() => {
        // Se reinicializa cada vez que se termina una prueba
        jest.clearAllMocks()
        wrapper = shallowMount( EntryList, {
            global: {
                mocks: {
                    $router: routerMock
                },
                plugins: [ store ]
            }
        })
    })
    test('Debe llamar al entrylistbyterm y mostrar las 3 entradas', () => { 
        expect( wrapper.findAll('entry-stub').length ).toBe(3)
        expect( wrapper.html ).toMatchSnapshot()
     })

     test('Debe de llamar el getEntriesByTerm y filtrar las entradas', async() => {
         const input = wrapper.find('input')
         await input.setValue('Soy un mock')
         expect(wrapper.findAll('entry-stub').length).toBe(1)
     })

     test('el boton de nuevo debe direccionar a /new', () => {
         wrapper.find('button').trigger('click')
         expect(routerMock.push).toHaveBeenCalledWith({
             name:'entry', params: {id: 'new'}
         })
     })
})