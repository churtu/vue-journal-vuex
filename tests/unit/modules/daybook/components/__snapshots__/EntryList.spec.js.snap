// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryList Debe llamar al entrylistbyterm y mostrar las 3 entradas 1`] = `[Function]`;
